### PR TITLE
[ENG-937] Add a filter for the plus-four portion of a zipcode.

### PR DESCRIPTION
**Pull request by heathdutton**
This is for clients that want ZIP+4, but want it in a separate field.

### CODE REVIEW

File: Helper/TokenHelper.php [hunk 1]
```diff
@@ -344,6 +346,21 @@ private function addHelper($type)
...
+                                }
+                                $start = 1 - (strlen($value) - $dash);
+                                $value = substr((string) $value, $start);
+                                if (4 !== strlen($value)) {
+                                    // Not precisely 4 digits, thus this is not a ZIP+4
+                                    return '';
+                                }
+
+                                return $value;
+                            },
```
Comment (dean-martin) [inline Helper/TokenHelper.php]: Consider using regex?
Maybe something like: 
```
preg_match("/\d-([\d]{4})/", $value, $matches);
if(empty($matches) {
    return '';
}

return $matches[1];
```

Comment (heathdutton) [reply Helper/TokenHelper.php]: Much better... was in too much of a hurry.

**Comment by heathdutton** [comment]
> Consider using RegEx, it's perfect for these types of situations.

done

File: Helper/TokenHelper.php [hunk 2]
```diff
@@ -344,6 +346,14 @@ private function addHelper($type)
 
                                 return $dash ? substr((string) $value, 0, $dash) : $value;
                             },
+                            'plus4' => function ($value) {
+                                preg_match('/\d-([\d]{4})/', $value, $matches);
```
Comment (westonwatson) [inline Helper/TokenHelper.php]: if this is for the delivery route digits, should it except both "-" and "+" ?

Comment (dean-martin) [reply Helper/TokenHelper.php]: \d+[-+]([\d]{4})

Comment (heathdutton) [reply Helper/TokenHelper.php]: I would say yes... but we don't have delivery route digits in our systems.

**Pull request merged**